lang/semgrep-grammars/src/semgrep-kotlin/grammar.js

```diff
@@ -74,13 +74,43 @@ module.exports = grammar(standard_grammar, {
 	    );
 	},
 
+  // We would like to be able to parse programs which have a newline between the
+  // class name and the constructor:
+  // class Foo
+  // constructor Bar() { ... }
+
+  // The problem is that the Kotlin parser inserts a semicolon after "Foo", making
+  // it such that we get interrupted in the middle of the class_declaration.
+  // To make it so we can continue, we allow everything after the class identifier
+  // to be a standalone statement in its own right. This way, we can parse both parts
+  // individually, and stitch them together at parsing time.
+
+  // We only need to amend statements here, because the consumers of _declaration are
+  // only class_member_declaration, top_level_object and _statement.
+  // The former has `secondary_constructor`, which already looks like what we want to
+  // add, and the second seems to be unused.
+  // So we just need to fix _statement.
```

Comment:
Good comment, but could you also include a note about what the non-hacky way to do this would be, perhaps with a link to an issue on the tree-sitter-kotlin repo? I assume the scanner would have to be updated to change the ASI behavior?

Reply:
Done.